commerce/domain/shipping/splitshipment/SplitShipmentSample.cls

```diff
@@ -0,0 +1,158 @@
+/**
+ * @description Custom Split Shipment Service sample.
+ */
+global class SplitShipmentSample extends CartExtension.SplitShipmentService {
+
+    public virtual override void arrangeItems(CartExtension.ItemArrangementRequest request) {
+        CartExtension.Cart cart = request.getCart();
+        Iterator<CartExtension.ItemArrange> itemArrangeIterator = request.getItemArrangeList().iterator();
+
+        // Get all productIds from the ItemArrange list
+        List<Id> productIds = new List<Id>();
+        while (itemArrangeIterator.hasNext()) {
+            CartExtension.ItemArrange itemArrange = itemArrangeIterator.next();
+            productIds.add(itemArrange.getProductId());
+        }
+
+        // Get conveyability info for each product
+        List<Product2> products = [
+            SELECT
+                Id
+                , IsConveyable__c
+            FROM Product2
+            WHERE Id IN :productIds
+        ];
+        Iterator<Product2> productsIterator = products.iterator();
+
+        Map<Id,Boolean> productConveyability = new Map<Id,Boolean>();
+        while (productsIterator.hasNext()) {
+            Product2 product = productsIterator.next();
+            productConveyability.put(product.Id, product.IsConveyable__c);
```

Comment:
Wold this fail `IsConveyable` custom field is not defined?

Reply:
Yes it would, I can add a try-catch for robustness.

Reply:
@pplat23 yes, actually Apex devs won't be able to save this file if the custom field is not yet defined in their org. I've added this in the Apex doc.

![Screenshot 2023-12-15 at 2 33 04 PM](https://github.com/forcedotcom/commerce-extensibility/assets/100236257/f2dbe561-4b21-42cb-b9fc-406638b7283f)